cdc/processor/pipeline/sink.go

```diff
@@ -334,16 +343,35 @@ func (n *sinkNode) HandleMessage(ctx context.Context, msg pipeline.Message) (boo
 			if err := n.flushSink(ctx, msg.PolymorphicEvent.CRTs); err != nil {
 				return false, errors.Trace(err)
 			}
-			atomic.StoreUint64(&n.resolvedTs, msg.PolymorphicEvent.CRTs)
+			err := n.retryFlushSink(ctx, event.CRTs)
+			if err != nil {
+				return false, errors.Trace(err)
+			}
+			atomic.StoreUint64(&n.resolvedTs, event.CRTs)
 			return true, nil
 		}
-		if err := n.emitEvent(ctx, event); err != nil {
+		start := time.Now()
+		err := retry.Do(ctx, func() error {
+			err := n.emitEvent(ctx, event)
+			cost := time.Since(start)
+			if cost >= 10*time.Second {
+				log.Warn("redo log buffer blocking too long", zap.Duration("blocking time(s)", cost))
```

Comment:
What's the frequency of the log? Does it flood when retrying?

Reply:
Yes, it would flood when retrying too many times. I will add some code to limit the warn log output.